src/Spectre.libGenetic.Tests/GenerationTest.cpp

```diff
@@ -0,0 +1,149 @@
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#define GTEST_LANG_CXX11 1
+
+#include <gtest/gtest.h>
+#include "Spectre.libGenetic/Generation.h"
+#include "Spectre.libGenetic/InconsistentChromosomeLengthException.h"
+#include <minwindef.h>
+
+namespace
+{
+using namespace Spectre::libGenetic;
+
+class GenerationInitializationTest: public ::testing::Test
+{
+protected:
+    const Individual smallerIndividual = Individual({ true, false, true });
+    const Individual trueIndividual = Individual({ true, true, true, true });
+    const Individual falseIndividual = Individual({ false, false, false, false });
+    const std::vector<Individual> inconsistentGeneration = { trueIndividual, smallerIndividual, trueIndividual };
+    const std::vector<Individual> generation1Data = { trueIndividual, trueIndividual, trueIndividual };
+    const std::vector<Individual> generation2Data = { falseIndividual, falseIndividual, falseIndividual, falseIndividual };
+};
+
+TEST_F(GenerationInitializationTest, initializes)
+{
+    EXPECT_NO_THROW(Generation(std::vector<Individual>(generation1Data)));
+}
+
+TEST_F(GenerationInitializationTest, throws_on_inconsistent_chromosome_lengths)
+{
+    EXPECT_THROW(Generation(std::vector<Individual>(inconsistentGeneration)), InconsistentChromosomeLengthException);
+}
+
+class GenerationTest : public GenerationInitializationTest
+{
+public:
+	GenerationTest():
+        generation1(std::vector<Individual>(generation1Data)),
+        generation2(std::vector<Individual>(generation2Data))
+    {}
+protected:
+	Generation generation1;
+	Generation generation2;
+
+	void SetUp() override
+	{
+		generation1 = Generation(std::vector<Individual>(generation1Data));
+		generation2 = Generation(std::vector<Individual>(generation2Data));
+	}
+};
+
+TEST_F(GenerationTest, give_proper_size)
+{
+    EXPECT_EQ(generation1.size(), generation1Data.size());
+    EXPECT_EQ(generation2.size(), generation2Data.size());
+}
+
+TEST_F(GenerationTest, addition_produces_generation_of_proper_size)
+{
+	const auto generation = generation1 + generation2;
+	const auto size = generation.size();
+	EXPECT_EQ(size, 7);
+}
+
+TEST_F(GenerationTest, addition_produces_generation_of_elements_from_first_and_then_second)
+{
+    const auto generation = Generation({ trueIndividual }) + Generation({ falseIndividual });
+    const auto first = generation[0];
+    const auto second = generation[1];
+    EXPECT_EQ(first[0], trueIndividual[0]);
+    EXPECT_EQ(second[0], falseIndividual[0]);
```

Comment:
How about introducing `operator==(const Individual& other)` for the purpose above? Could the user benefit from this?